flow/templates/bridges2.sh

```diff
@@ -1,56 +1,57 @@
 {# Templated in accordance with: https://www.psc.edu/resources/bridges-2/user-guide #}
 {% extends "slurm.sh" %}
 {% block tasks %}
-{% set threshold = 0 if force else 0.9 %}
-{% set cpu_tasks = operations|calc_tasks('np', parallel, force) %}
-{% set gpu_tasks = operations|calc_tasks('ngpu', parallel, force) %}
-{% if gpu_tasks %}
-{% if not ('GPU' in partition or force) %}
-{% raise "GPU operations require a GPU partition!" %}
-{% endif %}
-{# GPU nodes have 8 NVIDIA V100-32GB SXM2 #}
-{% set nn_gpu = gpu_tasks|calc_num_nodes(8) %}
-{% set nn = nn_gpu %}
-{% else %}
-{% if 'GPU' in partition and not force %}
-{% raise "Requesting GPU partition, but no GPUs requested!" %}
-{% endif %}
-{% set nn = nn|default(cpu_tasks|calc_num_nodes(128), true) %}
-{% endif %}
-{% if 'GPU' in partition %}
-{% set gpus_per_node = (gpu_tasks / nn)|round(0, 'ceil')|int %}
-{% set cpus_per_node = (cpu_tasks / nn)|round(0, 'ceil')|int %}
-{% if cpus_per_node > gpus_per_node * 5 and not force %}
-{% raise "Cannot request more than 5 CPUs per GPU." %}
-{% endif %}
-{% endif %}
-{% if partition == 'GPU' %}
+    {%- set threshold = 0 if force else 0.9 %}
+    {%- set cpu_tasks = operations|calc_tasks('np', parallel, force) %}
+    {%- set gpu_tasks = operations|calc_tasks('ngpu', parallel, force) %}
+    {%- if gpu_tasks %}
+        {%- if not ('GPU' in partition or force) %}
+            {%- raise "GPU operations require a GPU partition!" %}
+        {%- endif %}
+        {#- GPU nodes have 8 NVIDIA V100-32GB SXM2 #}
+        {%- set nn_gpu = gpu_tasks|calc_num_nodes(8) %}
+        {%- set nn = nn_gpu %}
+        {%- else %}
+        {%- if 'GPU' in partition and not force %}
+            {%- raise "Requesting GPU partition, but no GPUs requested!" %}
+        {%- endif %}
+        {%- set nn = nn|default(cpu_tasks|calc_num_nodes(128), true) %}
+    {%- endif %}
+    {%- if 'GPU' in partition %}
+        {%- set gpus_per_node = (gpu_tasks / nn)|round(0, 'ceil')|int %}
+        {%- set cpus_per_node = (cpu_tasks / nn)|round(0, 'ceil')|int %}
+        {%- if cpus_per_node > gpus_per_node * 5 and not force %}
+            {%- raise "Cannot request more than 5 CPUs per GPU." %}
+        {%- endif %}
+    {%- endif %}
+    {%- if partition == 'GPU' %}
 #SBATCH -N {{ nn|check_utilization(gpu_tasks, 8, threshold, 'GPU') }}
 #SBATCH --ntasks-per-node={{ cpus_per_node }}
 #SBATCH --gpus={{ gpu_tasks }}
-{% elif partition == 'GPU-shared' %}
+    {%- elif partition == 'GPU-shared' %}
 #SBATCH -N {{ nn|check_utilization(gpu_tasks, 1, threshold, 'GPU') }}
 #SBATCH --ntasks-per-node={{ cpus_per_node }}
 #SBATCH --gpus={{ gpu_tasks }}
```

Comment:
should we indent the elif portions too?

Also, what does the dash after the percent sign mean?

Reply:
The dash after the percent sign controls whitespace, allowing us to indent the jinja logic for readability but not the actual script lines. See:
- https://stackoverflow.com/questions/11813422/what-does-this-in-jinja2-template-engine-do
- https://jinja.palletsprojects.com/en/2.11.x/templates/#whitespace-control

Reply:
@cbkerr I found a better way to do this with `lstrip_blocks=True` instead of the awkward `{%-` syntax. It's also more correct -- the whitespace removal of `{%-` was too aggressive in some places.